pkg/build/builder/docker.go

```diff
@@ -416,6 +426,49 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV instruction with the environment variables
+// from env after every FROM instruction in the given Dockerfile instructions
+// and returns a new Dockerfile.
+func insertEnvAfterFrom(instructions []byte, env []kapi.EnvVar) ([]byte, error) {
+	if len(env) == 0 {
+		return instructions, nil
+	}
+	node, err := parser.Parse(bytes.NewReader(instructions))
+	if err != nil {
+		return nil, err
+	}
+
+	// buildEnv is an ENV Dockerfile instruction to define all environment
+	// variables from env.
+	buildEnv := []byte("ENV")
```

Comment:
Hmm, when constructing a Dockerfile programatically, the name of instructions are by far the least problematic part.
Before pushing this code here I experimented with actually rewriting every Dockerfile instruction using one canonical syntax for each instruction type. That was a lot of code that died in favor of the simpler `ParseTreeToDockerfile`.

In this dead code, I had to take care of how to write things like `ENV A=B`, but also `ENV FOO="value with spaces"`, and handle weird characters (e.g. \u0000, "), etc.

IMHO its the part below these comments here that deserve more attention. The objective there is to ensure we write:

```Dockerfile
ENV "key1"="value1" "believe me"="env var can have spaces in the name" "last"="value\"with quote"
```